python/ray/experimental/async_plasma.py

```diff
@@ -1,179 +1,13 @@
 import asyncio
-import ctypes
-import sys
-
-import pyarrow.plasma as plasma
 
 import ray
 from ray.services import logger
-
-INT64_SIZE = ctypes.sizeof(ctypes.c_int64)
-
-
-def _release_waiter(waiter, *_):
-    if not waiter.done():
-        waiter.set_result(None)
-
-
-class PlasmaProtocol(asyncio.Protocol):
-    """Protocol control for the asyncio connection."""
-
-    def __init__(self, plasma_client, plasma_event_handler):
-        self.plasma_client = plasma_client
-        self.plasma_event_handler = plasma_event_handler
-        self.transport = None
-        self._buffer = b""
-
-    def connection_made(self, transport):
-        self.transport = transport
-
-    def data_received(self, data):
-        self._buffer += data
-        messages = []
-        i = 0
-        while i + INT64_SIZE <= len(self._buffer):
-            msg_len = int.from_bytes(self._buffer[i:i + INT64_SIZE],
-                                     sys.byteorder)
-            if i + INT64_SIZE + msg_len > len(self._buffer):
-                break
-            i += INT64_SIZE
-            segment = self._buffer[i:i + msg_len]
-            i += msg_len
-            (object_ids, object_sizes,
-             metadata_sizes) = self.plasma_client.decode_notifications(segment)
-            assert len(object_ids) == len(object_sizes) == len(metadata_sizes)
-            for j in range(len(object_ids)):
-                messages.append((object_ids[j], object_sizes[j],
-                                 metadata_sizes[j]))
-
-        self._buffer = self._buffer[i:]
-        self.plasma_event_handler.process_notifications(messages)
-
-    def connection_lost(self, exc):
-        # The socket has been closed
-        logger.debug("PlasmaProtocol - connection lost.")
-
-    def eof_received(self):
-        logger.debug("PlasmaProtocol - EOF received.")
-        self.transport.close()
+from collections import defaultdict
 
 
 class PlasmaObjectFuture(asyncio.Future):
```

Comment:
Do we even need this anymore if it's an empty wrapper?

Reply:
Yes, it is used by async_compat to determine whether the result is being loaded from plasma or in-memory.